### PR TITLE
SKU to key lookup in Catalog

### PR DESCRIPTION
**Pull request by SpaceKatt**
There exist cases where one would want to lookup the specific `key` for a given `SKU`.

This PR adds a function to the catalog to perform this lookup

### CODE REVIEW

File: src/catalog/catalog.ts [hunk 1]
```diff
@@ -107,6 +118,14 @@ export class Catalog implements ICatalog {
         return this.mapSpecific.has(key);
     }
 
+    getKeyForSku(sku: SKU): Key {
+        const key = this.skuToKey.get(sku);
+        if (!key) {
```
Comment (tushitas) [inline src/catalog/catalog.ts]: do you want to use this.skuToKey.has(sku)  method to find out if the entry exists before retrieving it.

File: src/catalog/catalog.ts [hunk 2]
```diff
@@ -5,13 +5,15 @@ import {
...
     PID,
+    SKU,
     SpecificTypedEntity,
 } from './interfaces';
 
 export class Catalog implements ICatalog {
     private mapGeneric = new Map<PID, GenericTypedEntity>();
     private mapSpecific = new Map<Key, SpecificTypedEntity>();
     private pidToKeys = new Map<PID, Key[]>();
+    private skuToKey = new Map<SKU, Key>();
```
Comment (drub0y) [inline src/catalog/catalog.ts]: I wonder if the initialization and build up of this map (maybe all maps) shouldn't be deferred until first call. It's not free in terms of start-up overhead and obviously has some memory costs too. 

Perhaps a premature optimization, but I'm just throwing it out there for consideration.

File: src/catalog/catalog.ts (hunk 1 above)
Comment (zanawar) [reply src/catalog/catalog.ts]: Since he's returning the key I don't think that's necessary.

File: src/catalog/catalog.ts (hunk 2 above)
Comment (zanawar) [reply src/catalog/catalog.ts]: I agree with this being a bit premature.

File: src/catalog/catalog.ts [hunk 3]
```diff
@@ -71,6 +73,15 @@ export class Catalog implements ICatalog {
...
             }
+
+            if (this.skuToKey.has(item.sku)) {
+                throw TypeError(
+                    `Catalog: encountered duplicate sku ${item.sku}.`
+                );
+            }
+
+            // Add SKU to key mapping
+            this.skuToKey.set(item.sku, item.key);
```
Comment (zanawar) [inline src/catalog/catalog.ts]: Somewhat optional: these changes to this method aren't tested directly.

**Pull request merged**